File: src/libs/geom_cpp/Shape.cc

#include "Shape.h"

// ==============================================================================
//
//  Construction
//
// ==============================================================================

/*!
  \param name       -- Name of the shape.
  \param dimensions -- Dimensionality of the shape.
 */
Shape::Shape (std::string const &name,
              unsigned int const &dimensions)
{
    itsName       = name;
    itsDimensions = dimensions;
}

// ==============================================================================
//
//  Operators
//
// ==============================================================================

/*!
  \param other -- Other Shape object to make copy from.
 */
Shape& Shape::operator= (Shape const &other)
{
    itsName       = other.itsName;
    itsDimensions = other.itsDimensions;
    itsNofFaces   = other.itsNofFaces;
    
    return *this;
}

File: src/libs/geom_ruby/Shape.rb

##
## Author:: Lars Baehren (lbaehren@gmail.com)
##
## Base class defining the common interface and functionality for all shapes.
##

class Shape

  ## Name of the shape
  attr_reader :itsName
  ## Dimensions of the shape
  attr_reader :itsDimensions

  ## Argumented constructor
  def initialize(name, dimensions)
    @itsName, @itsDimensions = name, dimensions
  end

  ## Comparison operator for sorting
  def <=>(shape) 
    @itsDimensions <=> shape.itsDimensions
  end

  ## Output operator to string
  def to_s
    "#@itsName (#@itsDimensions)"
  end

  ## Get the name of the shape
  def name
    @itsName
  end

  ## Set the name of the shape
  def setName (name)
    @itsName = name
  end

  ## Get the dimensions of the shape
  def dimensions
    @itsDimensions
  end

  ## Set the dimensions of the shape
  def setDimensions (dim)
    @itsDimensions = dim
  end

end
